platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Telemetry.java

```diff
@@ -1,75 +1,56 @@
 package com.mapbox.mapboxsdk.maps;
 
-
-import com.mapbox.android.telemetry.MapboxTelemetry;
 import com.mapbox.android.telemetry.SessionInterval;
-import com.mapbox.android.telemetry.TelemetryEnabler;
-import com.mapbox.mapboxsdk.BuildConfig;
-import com.mapbox.mapboxsdk.Mapbox;
+import com.mapbox.mapboxsdk.module.telemetry.TelemetryImpl;
 
+/**
+ * @deprecated use {@link TelemetryImpl} instead.
+ */
+@Deprecated
 public class Telemetry {
-  static final String TWO_FINGER_TAP = "TwoFingerTap";
-  static final String DOUBLE_TAP = "DoubleTap";
-  static final String SINGLE_TAP = "SingleTap";
-  static final String PAN = "Pan";
-  static final String PINCH = "Pinch";
-  static final String ROTATION = "Rotation";
-  static final String PITCH = "Pitch";
-  private MapboxTelemetry telemetry;
-
-  private Telemetry() {
-    telemetry = new MapboxTelemetry(Mapbox.getApplicationContext(), Mapbox.getAccessToken(),
-      BuildConfig.MAPBOX_EVENTS_USER_AGENT);
-    TelemetryEnabler.State telemetryState = TelemetryEnabler.retrieveTelemetryStateFromPreferences();
-    if (TelemetryEnabler.State.ENABLED.equals(telemetryState)) {
-      telemetry.enable();
-    }
-  }
-
-  public static void initialize() {
```

Comment:
Thi might not be impactful but breaks semver. How about leaving an empty impl instead?

Reply:
great catch!